src/engine/source/hlp2/src/parse_field.cpp

```diff
@@ -90,30 +90,34 @@ void updateDoc(json::Json& doc,
                std::string_view hdr,
                std::string_view val,
                bool is_escaped,
-               std::string_view escape)
+               std::string_view escape,
+               bool is_quoted)
 {
     if (val.empty())
     {
         doc.setNull(hdr);
         return;
     }
 
-    int64_t i;
-    auto [ptr, ec] {utils::from_chars(val.data(), val.data() + val.size(), i)};
-    if (std::errc() == ec && (val.data() + val.size()) == ptr)
+    if(!is_quoted)
     {
-        doc.setInt64(i, hdr);
-        return;
-    }
-    else
-    {
-        double_t d;
-        auto [ptr, ec] {utils::from_chars(val.data(), val.data() + val.size(), d)};
+        int64_t i;
+        auto [ptr, ec] {utils::from_chars(val.data(), val.data() + val.size(), i)};
         if (std::errc() == ec && (val.data() + val.size()) == ptr)
         {
-            doc.setDouble(d, hdr);
+            doc.setInt64(i, hdr);
             return;
         }
+        else
+        {
+            double_t d;
+            auto [ptr, ec] {utils::from_chars(val.data(), val.data() + val.size(), d)};
```

Comment:
Do not re-declare variables in a smaller scope if the name was used in the higher scope, this makes it difficult to read the scope and is very error prone.